### PR TITLE
Bugfix/6484

### LINKED ISSUE #6484: Block explorer protos provides inadequate REST information
Corestate sets the following properties:

https://github.com/vegaprotocol/vega/blob/74e5fdc9a8bfe68b314212bf83c6359d6d853386/protos/sources/vega/api/v1/corestate.proto#L14-L21

While they're imperfect (lb.testnet.vega.xyz is no longer a valid server), they do at least:
- Set a version that represents that Vega version
- Sets a server that can be used to access the API

I've set a bonus task, which is to actually use a valid server. However the docs scripts already replace the invalid server with a valid one, and the docs repo is currently the only known downstream client of the protos, so an invalid server is fine *for now*. If the bonus task is too much work, create an alternate ticket. None of these are blocking, as I fix them downstream either manually or automatically anyway.

# Tasks
- [ ] Add openapiv2_swagger Name and Version properties to Block Explorer proto
- [ ] Add servers property with invalid server to block explorer proto

# Bonus task
- [ ] Work out how to populate the server field with a single valid server

### PR DESCRIPTION
**Pull request by jeremyletang**
close #6484

### CODE REVIEW

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 1]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
+  // The transaction corresponding to the hash.
   Transaction transaction = 1;
 }
 
 message ListTransactionsRequest {
-    uint32 limit = 1;
-    optional string before = 2;
-    optional string after = 3;
-    map<string, string> filters =4;
+  // The amount of transaction to be returned from the blockchain
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:42]:
```suggestion
  // The number of transactions to be returned from the blockchain
```

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 2]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
-    uint32 code = 6;
-    string cursor = 7;
-    vega.commands.v1.InputData command = 8;
+  // The height of the block the transaction was found
+  uint64 block = 1;
+  // The index of the transaction in the block
+  uint32 index = 2;
+  // The hash of the transaction
+  string hash = 3;
+  // The submitter of the transaction (vega public key)
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:64]:
```suggestion
  // The submitter of the transaction (Vega public key)
```

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 3]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
-    rpc GetTransaction(GetTransactionRequest) returns (GetTransactionResponse) {}
-    rpc ListTransactions(ListTransactionsRequest) returns (ListTransactionsResponse) {}
+  // Get transaction
+  //
+  // Get a transaction from the vega blockchain
+  rpc GetTransaction(GetTransactionRequest) returns (GetTransactionResponse) {}
+
+  // List transactions
+  //
+  // List transactions from the vega blockchain
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:27]:
```suggestion
  // List transactions from the Vega blockchain
```

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 4]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
+  schemes: [HTTP, HTTPS]
+  host: "lb.testnet.vega.xyz"
+};
 
 service BlockExplorerService {
-    rpc GetTransaction(GetTransactionRequest) returns (GetTransactionResponse) {}
-    rpc ListTransactions(ListTransactionsRequest) returns (ListTransactionsResponse) {}
+  // Get transaction
+  //
+  // Get a transaction from the vega blockchain
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:22]:
```suggestion
  // Get a transaction from the Vega blockchain
```

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 5]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
+
+option go_package = "code.vegaprotocol.io/vega/protos/blockexplorer/api/v1";
+
+option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+  info: {
+    title: "Vega block explorer APIs";
+    version: "v0.59.0+dev";
+  }
+  schemes: [HTTP, HTTPS]
+  host: "lb.testnet.vega.xyz"
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:16]: Isn't the load balancer gone?

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 6]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
+  rpc ListTransactions(ListTransactionsRequest) returns (ListTransactionsResponse) {}
  }
 
 message GetTransactionRequest {
+  // The hash of the transaction
   string hash = 1;
 }
 
 message GetTransactionResponse{
+  // The transaction corresponding to the hash.
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:37]:
```suggestion
  // The transaction corresponding to the hash
```

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 7]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
-    optional string before = 2;
-    optional string after = 3;
-    map<string, string> filters =4;
+  // The amount of transaction to be returned from the blockchain
+  uint32 limit = 1;
+  // An optional cursor to paginate the request
+  optional string before = 2;
+  // An optional cursor to paginate the request
+  optional string after = 3;
+  // Filters to apply to the request.
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:48]:
```suggestion
  // Filters to apply to the request
```

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 8]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
 message Transaction {
-    uint64 block = 1;
-    uint32 index = 2;
-    string hash = 3;
-    string submitter = 4;
-    string type = 5;
-    uint32 code = 6;
-    string cursor = 7;
-    vega.commands.v1.InputData command = 8;
+  // The height of the block the transaction was found
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:58]:
```suggestion
  // The height of the block the transaction was found in
```

File: protos/sources/blockexplorer/blockexplorer.proto [hunk 9]
```diff
@@ -2,42 +2,75 @@ syntax = "proto3";
...
+  string type = 5;
+  // The results code of the transaction (0 is success)
+  uint32 code = 6;
+  // The cursor for this transaction (in the page, used for paginated results)
+  string cursor = 7;
+  // The actual command of the transaction
+  vega.commands.v1.InputData command = 8;
 }
 
 // Copypasta [there must be a better way?] from
```
Comment (candida-d) [inline protos/sources/blockexplorer/blockexplorer.proto:76]: well this is intriguing

Comment (edd) [reply protos/sources/blockexplorer/blockexplorer.proto:76]: 🍝

File: protos/sources/blockexplorer/blockexplorer.proto (hunk 5 above)
Comment (jeremyletang) [reply protos/sources/blockexplorer/blockexplorer.proto:16]: I think it is, but I didn't know what to point it to to be fair.

File: protos/sources/blockexplorer/blockexplorer.proto (hunk 9 above)
Comment (jeremyletang) [reply protos/sources/blockexplorer/blockexplorer.proto:76]: this is being removed in a separate PR.

File: protos/sources/blockexplorer/blockexplorer.proto (hunk 5 above)
Comment (candida-d) [reply protos/sources/blockexplorer/blockexplorer.proto:16]: I don't either, but if it's non-functional that probably won't help. Maybe n06?

**Review by ValentinTrinque — DISMISSED** [review]

**Review by AnExsomnis — APPROVED** [review]

**Pull request merged**